packages/application-shell/src/apollo-links/token-retry-link.ts

```diff
@@ -7,22 +8,17 @@ import {
   MC_API_SUPPORTED_HEADERS,
 } from '@commercetools-frontend/constants';
 
-type ApolloContext = {
-  headers: { [key: string]: string };
-  skipTokenRetry?: boolean;
-};
-
 // This link retries requests to the CTP API that have been rejected
 // because of an invalid/expired oauth token.
 // To do so, we resend the request with the header "X-Force-Token: true"
 // so that the MC BE can issue a new token.
 // NOTE: the retry is not meant to work for the MC access token.
 
 export const getDoesGraphQLTargetSupportTokenRetry = (
-  context: ApolloContext
+  context: TApolloContext
 ): boolean => {
-  const target = (context.headers[MC_API_SUPPORTED_HEADERS.TOKEN_RETRY] ||
-    context.headers[
+  const target = (context.headers?.[MC_API_SUPPORTED_HEADERS.TOKEN_RETRY] ||
+    context.headers?.[
```

Comment:
Headers are considered optional according to usage elsewhere. To make shared type work be have to follow it here.

Reply:
Right, so I guess the type should be `TTokenRetryGraphQlTarget | undefined`